src/main/java/com/onarandombox/MultiverseCore/commands/DumpsCommand.java

```diff
@@ -0,0 +1,320 @@
+                            }
+                        })
+                        .completion(() -> {
+                            List<String> types = new ArrayList<>();
+                            for (Services type : Services.values()) {
+                                types.add(type.name().toLowerCase());
+                            }
+                            return types;
+                        })
+                        .build())
+                .add(CommandFlag.builder("--paranoid")// Does not upload logs or plugin list (except if --logs mclogs is there)
+                        .addAlias("-p")
+                        .build())
+                .build());
+    }
+
+    private enum Services {
+        PASTEGG,
+        PASTESDEV
+    }
+
+    private enum LogsType {
+        APPEND,
+        MCLOGS
+    }
+
+    @Subcommand("dumps")
+    @CommandPermission("multiverse.core.dumps")
+    @CommandCompletion("@flags:groupName=mvdumps")
+    @Syntax("--logs [mclogs|append] --upload [pastesdev|pastegg --paranoid")
+    @Description("{@@mv-core.dumps.description}")
+    public void onDumpsCommand(CommandIssuer issuer,
+
+                               @Optional
+                               @Syntax("--logs [mclogs|append] --upload [pastesdev|pastegg] --paranoid")
+                               String[] flags
+    ) {
+        ParsedCommandFlags parsedFlags = parseFlags(flags);
+
+        MVVersionEvent versionEvent = new MVVersionEvent();
+
+        this.addDebugInfoToEvent(versionEvent);
+        plugin.getServer().getPluginManager().callEvent(versionEvent);
+
+        // Add plugin list if user isn't paranoid
+        if (!parsedFlags.hasFlag("--paranoid")) {
+            versionEvent.putDetailedVersionInfo("plugins.md", "# Plugins\n\n" + getPluginList());
+        }
+
+
+        // Deal with default case of logs
+        LogsType logsType = parsedFlags.flagValue("--logs", LogsType.class);
+        if (logsType == null) {
+            logsType = LogsType.MCLOGS;
+            Logging.finer("Logs was null so set to mclogs");
+        }
+
+
+
+
+        // Deal with default case of upload
+        Services services = parsedFlags.flagValue("--upload", Services.class);
+        if (services == null) {
+            services = Services.PASTEGG;
+            Logging.finer("Upload was null so set to pastegg");
+        }
+
+
+        // Need to be final for some reason...
+        final LogsType finalLogsType = logsType;
+        final Services finalServices = services;
+        BukkitRunnable logPoster = new BukkitRunnable() {
+            @Override
+            public void run() {
+                HashMap<String, String> pasteURLs = new HashMap<>();
+                Logging.finer("Logs type is: " + finalLogsType);
+                Logging.finer("Services is: " + finalServices);
+
+                // Deal with logs flag
+                if (!parsedFlags.hasFlag("--paranoid")) {
```

Comment:
I think this check should be done above with the `// Deal with default case of logs`